adam/language_specific/english/english_language_generator.py

```diff
@@ -220,6 +209,63 @@ def _translate_object_to_noun(
 
             return dependency_node
 
+        def add_determiner(
+            self,
+            _object: SituationObject,
+            count: int,
+            dependency_node: DependencyTreeToken,
+        ) -> None:
+            possession_relations = [
+                relation
+                for relation in self.situation.relations
+                if relation.relation_type == HAS and relation.second_slot == _object
+            ]
+            if len(possession_relations) > 1:
+                raise RuntimeError("Can not handle multiple possession relations")
+            else:
+                # If speaker possesses the noun
+                if (
+                    len(possession_relations) == 1
+                    and IS_SPEAKER in possession_relations[0].first_slot.properties
+                ):
+                    quantifier_node = DependencyTreeToken("my", DETERMINER)
+                    quantifier_role = DETERMINER_ROLE
+                # If addressee possess the noun
+                elif (
+                    len(possession_relations) == 1
+                    and IS_ADDRESSEE in possession_relations[0].first_slot.properties
+                ):
+                    quantifier_node = DependencyTreeToken("your", DETERMINER)
+                    quantifier_role = DETERMINER_ROLE
+                # add 's if a non-agent possess the noun
+                elif (
+                    len(possession_relations) == 1
+                    and possession_relations[0].first_slot
+                    not in only(self.situation.actions).argument_roles_to_fillers[AGENT]
+                ):
+                    token = (
+                        self.objects_to_dependency_nodes[
+                            possession_relations[0].first_slot
+                        ].token
+                        + "'s"
```

Comment:
I think most dependency trees analyze `'s` as its own token, attached to the possessor by a `CASE` role.